File: src/main/java/net/dontdrinkandroot/gitki/wicket/security/Enable.java
package net.dontdrinkandroot.gitki.wicket.security;

import net.dontdrinkandroot.gitki.model.Role;

import java.lang.annotation.*;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PACKAGE, ElementType.TYPE})
@Documented
@Inherited
public @interface Enable
{
    Role value();
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/security/Enable.kt
package net.dontdrinkandroot.gitki.wicket.security

import net.dontdrinkandroot.gitki.model.Role
import java.lang.annotation.Inherited

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FILE, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@MustBeDocumented
@Inherited
annotation class Enable(val value: Role)